src/main/resources/assets/opencomputers/loot/openos/bin/sh.lua

```diff
@@ -41,7 +41,16 @@ if #args == 0 then
   end
 else
   -- execute command.
-  local result = table.pack(sh.execute(...))
+  local cargs = table.pack(...)
+  -- sh can run as a shell command (no env table)
+  local cenv = _ENV
+  local cargsStart = 1
+  if type(cargs[1]) == "table" then
+    -- sh can also run as a manually started process (see /bin/source.lua)
+    cenv = cargs[1]
+    cargsStart = 2
+  end
+  local result = table.pack(sh.execute(cenv, table.unpack(cargs, cargsStart)))
```

Comment:
This could be shortened a bit.
```lua
  local cenv = _ENV
  if type(cargs[1]) == "table" then
    -- sh can also run as a manually started process (see /bin/source.lua)
    cenv = table.remove(cargs, 1)
  end
  local result = table.pack(sh.execute(cenv, table.unpack(cargs)))
```

Also, what about using `{...}` instead of `table.pack()`?

Reply:
Same as with using `table.unpack`, I figured I'd just stick to what's already being used and used `table.pack` instead of `{...}`, mixing multiple ways to achieve the same thing in the same chunk of code.

Reply:
They are not completely identical. Using `{...}` would pack all arguments into a table, whereas `table.pack` actually also adds the additional key `"n"`.

Reply:
TIL 😅 anyway, I've just pushed another commit implementing your proposed changes.

Reply:
This is up to @payonel anyway.

Reply:
This is up to payonel anyway.